ehcache/README.md

```diff
@@ -0,0 +1,86 @@
+# templates-ehcache
+
+## Summary
+Template / Example for a microservice using spring cache abstraction and EHCACHE library.
+
+Spring Cache Abstraction: https://docs.spring.io/spring-framework/docs/3.2.x/spring-framework-reference/html/cache.html
+
+EHCACHE: https://www.ehcache.org/
+
+## Use case
+
+For device / asset integration the external id (unique external identifier) is used for correlate devices and assets.
+To keep the communication less frequent to the platform a managed cache can be used. Be aware server side (microserivce)
+integration has fewer problems with frequent platform communication. This approach is more interesting for agent and edge development, where bandwidth to cloud isn't always high. 
+
+## How to run locally:
+
+1. Build project
+
+mvn clean install
+
+2. Create application at comulocity
+
+POST 'https://{base-url}/application/applications'
+
+Request body:
+
+```javascript
+{
+  "key": "templates-ehcache",
+  "name": "templates-ehcache",
+  "contextPath": "templates-ehcache",
+  "type": "MICROSERVICE",
+  "manifest":{},	
+	"requiredRoles": [
+		"ROLE_INVENTORY_READ",
+		"ROLE_INVENTORY_CREATE",
+		"ROLE_INVENTORY_ADMIN",
+		"ROLE_IDENTITY_READ",
+		"ROLE_IDENTITY_ADMIN"
+	],
+	"roles": [
+	]
+}
+```
+
+3. Subscribe application to your tenant via UI
+
+4. Acquire microservice bootstrap credentials
+
+GET 'https://{base-url}/application/applications/{applicationId}/bootstrapUser
+
+Response body:
+
+```javascript
+{
+    "password": "************************",
+    "name": "servicebootstrap_templates-ehcache",
+    "tenant": "<your tenant>"
+}
+```
+
+5. Add bootstrap credentials 
+
+to your src/main/resources/application-dev.properties
+
+6. Set environment variable
+
+C8Y_MICROSERVICE_ISOLATION=MULTI_TENANT
```

Comment:
Setting the environment variable can be removed. This shouldn't be necessary anymore with the latest version of the SDK

Reply:
Yes you are right, but it didn't work for me, if environment is not set, microservice is not starting.

Reply:
I checked again: 

         _                                    _          
          (_)                                  (_)         
 _ __ ___  _  ___ _ __ ___  ___  ___ _ ____   ___  ___ ___ 
| '_ ` _ \| |/ __| '__/ _ \/ __|/ _ \ '__\ \ / / |/ __/ _ \
| | | | | | | (__| | | (_) \__ \  __/ |   \ V /| | (_|  __/
|_| |_| |_|_|\___|_|  \___/|___/\___|_|    \_/ |_|\___\___|
                                                           
                                                                  
Version (0.0.1-SNAPSHOT)

 :: Cumulocity Microservices SDK (1006.11.0) ::                                                                                                                
 :: Spring Boot (v2.2.4.RELEASE) ::
 
2020-11-12 12:49:53.855  INFO 14608 --- [           main] com.c8y.ms.templates.basic.App           : Starting App on SAG-90FWMH2 with PID 14608 (C:\Users\APES\Development\workspace51\cumulocity-microservice-templates\basic\target\classes started by APES in C:\Users\APES\Development\workspace51\cumulocity-microservice-templates)
2020-11-12 12:49:53.859  INFO 14608 --- [           main] com.c8y.ms.templates.basic.App           : The following profiles are active: dev
2020-11-12 12:49:55.898  INFO 14608 --- [           main] o.s.c.a.ConfigurationClassEnhancer       : @Bean method EnableContextSupportConfiguration.contextScopeConfigurer is non-static and returns an object assignable to Spring's BeanFactoryPostProcessor interface. This will result in a failure to process annotations such as @Autowired, @Resource and @PostConstruct within the method's declaring @Configuration class. Add the 'static' modifier to this method to avoid these container lifecycle issues; see @Bean javadoc for complete details.
2020-11-12 12:49:56.193  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.security.config.annotation.configuration.ObjectPostProcessorConfiguration' of type [org.springframework.security.config.annotation.configuration.ObjectPostProcessorConfiguration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:56.196  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'objectPostProcessor' of type [org.springframework.security.config.annotation.configuration.AutowireBeanFactoryObjectPostProcessor] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:56.198  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.security.access.expression.method.DefaultMethodSecurityExpressionHandler@1f53481b' of type [org.springframework.security.access.expression.method.DefaultMethodSecurityExpressionHandler] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:56.211  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'com.cumulocity.microservice.security.annotation.UserDetailsServiceConfiguration' of type [com.cumulocity.microservice.security.annotation.UserDetailsServiceConfiguration$$EnhancerBySpringCGLIB$$a69c68de] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:56.275  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'applicationApi' of type [org.springframework.aop.scope.ScopedProxyFactoryBean] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:56.275  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'applicationApi' of type [com.cumulocity.microservice.subscription.repository.application.ApplicationApi$$EnhancerBySpringCGLIB$$8c839f45] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:56.283  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'securityExpressionService' of type [com.cumulocity.microservice.security.service.impl.SecurityExpressionServiceImpl] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:56.283  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'com.cumulocity.microservice.security.annotation.EnableGlobalMethodSecurityConfiguration' of type [com.cumulocity.microservice.security.annotation.EnableGlobalMethodSecurityConfiguration$$EnhancerBySpringCGLIB$$f401f735] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:56.298  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.security.config.annotation.method.configuration.Jsr250MetadataSourceConfiguration' of type [org.springframework.security.config.annotation.method.configuration.Jsr250MetadataSourceConfiguration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:56.303  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'jsr250MethodSecurityMetadataSource' of type [org.springframework.security.access.annotation.Jsr250MethodSecurityMetadataSource] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:56.306  INFO 14608 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'methodSecurityMetadataSource' of type [org.springframework.security.access.method.DelegatingMethodSecurityMetadataSource] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2020-11-12 12:49:57.230  INFO 14608 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2020-11-12 12:49:57.257  INFO 14608 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2020-11-12 12:49:57.258  INFO 14608 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.30]
2020-11-12 12:49:57.459  INFO 14608 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2020-11-12 12:49:57.459  INFO 14608 --- [           main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 3451 ms
2020-11-12 12:49:57.591  INFO 14608 --- [           main] bleMicroserviceSubscriptionConfiguration : Microservice repository will be build for application 'templates-basic'.
2020-11-12 12:49:58.070  INFO 14608 --- [           main] c.c.m.s.a.TokenCacheConfiguration        : Default Guava implementation for token cache is used.
Parameters:
- max cache size: 10000
- expire after access: 10 minutes
2020-11-12 12:49:58.105  INFO 14608 --- [           main] o.s.boot.web.servlet.RegistrationBean    : Filter preAuthenticateServletFilter was not registered (disabled)
2020-11-12 12:49:58.105  INFO 14608 --- [           main] o.s.boot.web.servlet.RegistrationBean    : Filter postAuthenticateServletFilter was not registered (disabled)
2020-11-12 12:49:58.319  INFO 14608 --- [           main] o.s.s.web.DefaultSecurityFilterChain     : Creating filter chain: any request, [org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter@2871ac91, org.springframework.security.web.header.HeaderWriterFilter@1a14fdf0, org.springframework.security.web.authentication.logout.LogoutFilter@4168f3d9, com.cumulocity.microservice.security.token.CumulocityOAuthMicroserviceFilter@1d3c112a, com.cumulocity.microservice.security.filter.PreAuthenticateServletFilter@37a9b687, org.springframework.security.web.authentication.www.BasicAuthenticationFilter@7aded903, org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter@435e416c, org.springframework.security.web.authentication.AnonymousAuthenticationFilter@3c54ddec, com.cumulocity.microservice.security.filter.PostAuthenticateServletFilter@525b1b70, org.springframework.security.web.access.ExceptionTranslationFilter@111c229c, org.springframework.security.web.access.intercept.FilterSecurityInterceptor@29d25e65]
2020-11-12 12:49:58.536  INFO 14608 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2020-11-12 12:49:58.896  INFO 14608 --- [           main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 4 endpoint(s) beneath base path ''
2020-11-12 12:49:58.966  INFO 14608 --- [           main] .s.s.i.MicroserviceSubscriptionScheduler : Start; subscriptionDelay = 10000, subscriptionInitialDelay = 30000
2020-11-12 12:49:58.994  INFO 14608 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2020-11-12 12:49:58.999  INFO 14608 --- [           main] com.c8y.ms.templates.basic.App           : Started App in 5.957 seconds (JVM running for 6.787)
2020-11-12 12:50:25.955  INFO 14608 --- [nio-8080-exec-1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2020-11-12 12:50:25.955  INFO 14608 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2020-11-12 12:50:25.966  INFO 14608 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 11 ms
2020-11-12 12:50:27.096  WARN 14608 --- [nio-8080-exec-1] c.c.m.s.r.i.LegacyMicroserviceRepository : User has no permission to api GET https://smartlabels.eu-latest.cumulocity.com/application/applicationsByName/templates-basic
2020-11-12 12:50:27.149  WARN 14608 --- [nio-8080-exec-1] c.c.m.s.r.i.LegacyMicroserviceRepository : User has no permission to api POST https://smartlabels.eu-latest.cumulocity.com/application/applications
2020-11-12 12:50:27.151  WARN 14608 --- [nio-8080-exec-1] c.c.m.context.ContextServiceImpl         : execution of task failed within tenant : smartlabels - Application templates-basic not found 
2020-11-12 12:50:27.161 ERROR 14608 --- [nio-8080-exec-1] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception

java.lang.IllegalStateException: Application templates-basic not found
	at com.cumulocity.microservice.subscription.service.impl.MicroserviceSubscriptionsServiceImpl.lambda$registerApplication$4(MicroserviceSubscriptionsServiceImpl.java:170) ~[microservice-subscription-1006.11.0.jar:na]
	at java.base/java.util.Optional.orElseThrow(Optional.java:408) ~[na:na]
	at com.cumulocity.microservice.subscription.service.impl.MicroserviceSubscriptionsServiceImpl.registerApplication(MicroserviceSubscriptionsServiceImpl.java:170) ~[microservice-subscription-1006.11.0.jar:na]
	at com.cumulocity.microservice.subscription.service.impl.MicroserviceSubscriptionsServiceImpl.subscribe(MicroserviceSubscriptionsServiceImpl.java:114) ~[microservice-subscription-1006.11.0.jar:na]
	at com.cumulocity.microservice.subscription.service.impl.MicroserviceSubscriptionsServiceImpl.getCredentials(MicroserviceSubscriptionsServiceImpl.java:215) ~[microservice-subscription-1006.11.0.jar:na]
	at com.cumulocity.microservice.security.filter.provider.SpringSecurityContextProvider.get(SpringSecurityContextProvider.java:55) ~[microservice-security-1006.11.0.jar:na]
	at com.cumulocity.microservice.security.filter.provider.SpringSecurityContextProvider.get(SpringSecurityContextProvider.java:18) ~[microservice-security-1006.11.0.jar:na]
	at com.cumulocity.microservice.security.filter.PostAuthenticateServletFilter$3.apply(PostAuthenticateServletFilter.java:61) ~[microservice-security-1006.11.0.jar:na]
	at com.cumulocity.microservice.security.filter.PostAuthenticateServletFilter$3.apply(PostAuthenticateServletFilter.java:59) ~[microservice-security-1006.11.0.jar:na]
	at com.google.common.collect.Iterators$7.transform(Iterators.java:750) ~[guava-21.0.jar:na]
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:47) ~[guava-21.0.jar:na]
	at com.google.common.collect.Iterators$6.computeNext(Iterators.java:616) ~[guava-21.0.jar:na]
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:145) ~[guava-21.0.jar:na]
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:140) ~[guava-21.0.jar:na]
	at com.google.common.collect.ImmutableList.copyOf(ImmutableList.java:267) ~[guava-21.0.jar:na]
	at com.google.common.collect.ImmutableList.copyOf(ImmutableList.java:229) ~[guava-21.0.jar:na]
	at com.google.common.collect.FluentIterable.toList(FluentIterable.java:609) ~[guava-21.0.jar:na]
	at com.cumulocity.microservice.security.filter.PostAuthenticateServletFilter.doFilterInternal(PostAuthenticateServletFilter.java:67) ~[microservice-security-1006.11.0.jar:na]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:158) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.authentication.www.BasicAuthenticationFilter.doFilterInternal(BasicAuthenticationFilter.java:203) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at com.cumulocity.microservice.security.filter.PreAuthenticateServletFilter$1.run(PreAuthenticateServletFilter.java:42) ~[microservice-security-1006.11.0.jar:na]
	at com.cumulocity.microservice.context.ContextServiceImpl$CallableRunnableWrapper.call(ContextServiceImpl.java:119) ~[microservice-context-1006.11.0.jar:na]
	at com.cumulocity.microservice.context.ContextServiceImpl$CallableRunnableWrapper.call(ContextServiceImpl.java:109) ~[microservice-context-1006.11.0.jar:na]
	at com.cumulocity.microservice.context.ContextServiceImpl.callWithinContext(ContextServiceImpl.java:78) ~[microservice-context-1006.11.0.jar:na]
	at com.cumulocity.microservice.context.ContextServiceImpl.runWithinContext(ContextServiceImpl.java:66) ~[microservice-context-1006.11.0.jar:na]
	at com.cumulocity.microservice.context.ContextServiceImpl$4.run(ContextServiceImpl.java:129) ~[microservice-context-1006.11.0.jar:na]
	at com.cumulocity.microservice.security.filter.PreAuthenticateServletFilter.doFilterInternal(PreAuthenticateServletFilter.java:75) ~[microservice-security-1006.11.0.jar:na]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at com.cumulocity.microservice.security.token.CumulocityOAuthMicroserviceFilter.doFilter(CumulocityOAuthMicroserviceFilter.java:89) ~[microservice-security-1006.11.0.jar:na]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:92) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:77) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:215) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:178) ~[spring-security-web-5.2.1.RELEASE.jar:5.2.1.RELEASE]
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:108) ~[spring-boot-actuator-2.2.4.RELEASE.jar:2.2.4.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119) ~[spring-web-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:367) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1598) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-embed-core-9.0.30.jar:9.0.30]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2020-11-12 12:50:29.022  WARN 14608 --- [subscriptions-0] c.c.m.s.r.i.LegacyMicroserviceRepository : User has no permission to api GET https://smartlabels.eu-latest.cumulocity.com/application/applicationsByName/templates-basic
2020-11-12 12:50:29.050  WARN 14608 --- [subscriptions-0] c.c.m.s.r.i.LegacyMicroserviceRepository : User has no permission to api POST https://smartlabels.eu-latest.cumulocity.com/application/applications